beetsplug/replaygain.py

```diff
@@ -312,10 +327,25 @@ def end_element_handler(name):
                                           'the output of bs1770gain')
                 album_gain["album"] = Gain(state['gain'], state['peak'])
                 state['gain'] = state['peak'] = None
+            elif len(per_file_gain) == len(path_list):
+                if state['gain'] is not None:
+                    album_state['gain'] = state['gain']
+                if state['peak'] is not None:
+                    album_state['peak'] = state['peak']
+                if album_state['gain'] is not None \
+                        and album_state['peak'] is not None:
+                    album_gain["album"] = Gain(
+                        album_state['gain'], album_state['peak'])
+                state['gain'] = state['peak'] = None
+
         parser.StartElementHandler = start_element_handler
         parser.EndElementHandler = end_element_handler
 
         try:
+            if type(text) == bytes:
+                text = text.decode('utf-8')
+            while '\x08' in text:
```

Comment:
It might be safer to loop until the string stops changing. Otherwise, this runs the risk of this character being present but the pattern not matching (e.g., if it occurs at the beginning of the string).

Reply:
To me it seems more readable to cover that case in the pattern by also replacing \x08 at the start of the string
```regex
b'[^\x08]\x08|^\x08'
```
then I can just explain it in a comment since I should write a bit about what's going on here anyway (: